builtin/logical/database/path_roles_test.go

```diff
@@ -1353,6 +1353,39 @@ func TestIsInsideRotationWindow(t *testing.T) {
 	}
 }
 
+// TestStaticRoleTTLAfterUpdate tests that a static roles
+// TTL is properly updated after updating rotation period
+// This addresses a bug in which NextVaultRotation was not
+// set on update
+func TestStaticRoleTTLAfterUpdate(t *testing.T) {
+	ctx := context.Background()
+	b, storage, mockDB := getBackend(t)
+	defer b.Cleanup(ctx)
+	configureDBMount(t, storage)
+
+	roleName := "hashicorp"
+	data := map[string]interface{}{
+		"username":        "hashicorp",
+		"db_name":         "mockv5",
+		"rotation_period": "10m",
+	}
+
+	createRoleWithData(t, b, storage, mockDB, roleName, data)
+	// read credential
+	resp := readStaticCred(t, b, storage, mockDB, roleName)
+	initialTTL := resp.Data["ttl"]
+
+	updateStaticRoleWithData(t, b, storage, mockDB, roleName, map[string]interface{}{
+		"username":        "hashicorp",
+		"db_name":         "mockv5",
+		"rotation_period": "20m",
+	})
+
+	resp = readStaticCred(t, b, storage, mockDB, roleName)
+	updatedTTL := resp.Data["ttl"]
+	require.NotEqual(t, initialTTL, updatedTTL)
```

Comment:
I wonder if comparing the equality of these two values is sufficient. Do we expect the updated TTL to greater than the previous TTL?

Reply:
Yes, the TTL should be different! I updated the role with a new rotation_period. It was from 10->20 minutes, so the TTL should update accordingly.

Reply:
Can you update the tests to reflect that. Asserting they are not equal may not be enough.